packages/mobile/src/account/saga.ts

```diff
@@ -82,6 +84,28 @@ function* initializeAccount() {
   }
 }
 
+export function* watchDailyLimit() {
+  const account = yield call(getConnectedAccount)
+  const channel = yield call(cUsdDailyLimitChannel, account)
+  if (!channel) {
+    return
+  }
+  try {
+    while (true) {
+      const dailyLimit = yield take(channel)
+      if (_.isNumber(dailyLimit)) {
```

Comment:
Why is this necessary? What else can `dailyLimit` be?

Reply:
Just a sanity check. If someone accidentaly writes a string or an object on Firebase this will prevent the app from failing.

Reply:
Can't we restrict the data type of the database to `integer`? Seems like it would be better to stop this as far upstream as possible.

Reply:
We will still need a check here to look for empty values since this is an optional field. We can still look into adding a check on Firebase, but I don't see the problem with this particular check here.

Reply:
I added a rule to firebase to restrict to numbers, but those restrictions don't apply to admin users or editing from the console, so there is still room for human error :( https://stackoverflow.com/questions/50658307/validate-firebase-with-rules-when-data-is-written-by-firebase-admin

Reply:
Cool! No problem with this specific check. What bothered me was that it fails silently so I was thinking about how we can prevent it all together.

What do you think about throwing an error if `dailyLimit` is defined but not a number? That way we can have some indication of what went wrong in the logs.

Reply:
I can add a log I guess? Throwing an error would be a bit extreme imo, how would you expect the error to be handled? Maybe fire an analytics event?

Reply:
It looks like throwing an error here will get caught and logged by the `catch` clause. I agree that displaying an error to the user is probably too much. We probably don't need analytics for this either but your call.

Reply:
Added a log rather than throwing the error so that we're still listening to the Firebase channel and receive updates if they come 😄